app/controllers/api/rss_channels_controller.rb

```diff
@@ -11,6 +11,11 @@ def create
       head :created
     end
 
+    def destroy
+      @url = Post.find_by(url: params[:url])
```

Comment:
```suggestion
      @channel = RssChannel.find(params[:id])
```

考え方は悪くないのですが、 `id` で特定した方が一意性が確実なのと、 `find_by` だとレコードが見つからなかった時に例外が発生しないので、多くの場合は `find` を使います。
`find` は引数にレコードの id を取り、見つからなかった場合は例外が発生します。
例外が発生したら、rails が自動的に 404 を返してくれます（多分）。

あと、 `Post` というリソースはありません。